src/extensions/default/Inline3DParametersEditor/ParameterEditor.js

```diff
@@ -0,0 +1,214 @@
+/*
+ * File based on InlineColorEditor/ColorEditor.jss
+ */
+
+ define(function (require, exports, module) {
+    "use strict";
+
+    var Strings                 = brackets.getModule("strings"),
+        Mustache                = brackets.getModule("thirdparty/mustache/mustache"),
+        Inline3dParametersUtils = require("Parameters3DUtils"),
+        Parameters              = require("text!Parameters.json");
+
+    /** Mustache template that forms the bare DOM structure of the UI */
+    var Template = require("text!ParameterEditorTemplate.html");
+
+    function ParameterEditor($parent, callback, tag, parameters) {
+        // Create the DOM structure, filling in localized strings via Mustache
+        this.$element = $(Mustache.render(Template, Strings));
+        $parent.append(this.$element);
+        this._callback = callback;
+        this._tag = tag;
+        this._modifier = Inline3dParametersUtils.getModifier(this._tag) || 10;
+
+        this._handleSliderDrag = this._handleSliderDrag.bind(this);
+
+        this.setSliderProperties(parameters);
+
+        this.$selectionBase = this.$element.find(".selector-base");
+
+        // Attach event listeners to main UI elements
+        this._addListeners();
+    }
+
+
+    /** Returns the root DOM node of the UI */
+    ParameterEditor.prototype.getRootElement = function () {
+        return this.$element;
+    };
+
+    /**Registers event listeners for sliders*/
+    ParameterEditor.prototype._addListeners = function () {
+        var self = this;
+        for(var i = 0; i < this._numberOfParameters; i++) {
+            this.$sliders[i].bind("mousedown", {"index" : i, "self" : this}, this._mouseDownCallback);
+            this._registerDragHandler(this.$sliders[i], this._handleSliderDrag, i);
+        }
+    };
+
+    ParameterEditor.prototype._mouseDownCallback = function(event) {
+        var index = event.data.index;
+        var self = event.data.self;
+        self._values[index] = parseFloat(event.currentTarget.value);
+        self._position = event.clientX;
+    };
+
+    ParameterEditor.prototype.focus = function () {
+        if (!this.$selectionBase.is(":focus")) {
+            this.$selectionBase.focus();
+            return true;
+        }
+        return false;
+    };
+
+    ParameterEditor.prototype._commitParameters = function (parameters) {
+        this._callback(parameters);
+    };
+
+    ParameterEditor.prototype.setParametersFromString = function (parameters) {
+        this.setSliderProperties(parameters);
+        this._commitParameters(parameters, true);
+    };
+
+    ParameterEditor.prototype.isValidSetOfParameters = function (parameters) {
+        var ParameterRegex = new RegExp(Inline3dParametersUtils.PARAMETERS_3D_REGEX);
```

Comment:
Yes that was fixed earlier. It wasn't a bug more how global regexes work. 
Fixed [here](https://github.com/mozilla/brackets/pull/651/commits/e6b13a5c55ad26b6ca0fd7c429ebcd6f7837aebd#diff-04be0ab1e56791052f3f8a21e93f50f0R37)